src/pages/_app.js

```diff
@@ -1,40 +1,41 @@
-import React from 'react'
-import App from 'next/app'
+import React from 'react';
+import googleAnalytics from '@asyncapistudio/plugin-google-analytics';
 import AppContext from '../contexts/AppContext'
 import '../css/tailwind.css'
+import clientEvents from '../lib/client-events';
 
-class AsyncApiStudio extends App {
-  static async getInitialProps({ ctx }) {
-    if (!ctx || !ctx.req) return {}
+function AsyncApiStudio({ Component, pageProps, context }) {
+  if (typeof window !== 'undefined') {
+    clientEvents.on('page:render', () => googleAnalytics.init(window));
+    clientEvents.emit('page:render');
+  }
+  return (
+    <AppContext.Provider value={context || {}}>
+      <Component {...pageProps} />
+    </AppContext.Provider>
+  );
+}
```

Comment:
@AceTheCreator I will answer here for you question. 

As I wrote, the code:

```ts
    clientEvents.on('page:render', () => googleAnalytics.init(window));
    clientEvents.emit('page:render');
```

will be only executed on the client side and you will lost all events binded to `clientEvents` registry. As you wrote in previous comments, `googleAnalytics.init` only works when you bind this plugin as `clientEvents.on('page:render', () => googleAnalytics.init(window));` in this file. So I see steps to do:

- load events from plugins to `clientEvents` registry. We must remember that they should be loaded on client side, so loading in this file is probably the best place, but:
  - to load events we use the file system. It doesn't exist on client side, so we must handle this problem on different way, I mean load events from external packages/plugins but without filesystem. We can use https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr to load dynamic modules on client side :) However there is an another problem, because we must know about names of external packages/plugins. So this is the hardest part. How to get the packages name, and then load events from them? If you don't understand, please see example:
  
    If we will use the dynamic import from nextJS, we will have something like:
    
    ```js
    const DynamicPlugin = dynamic(
      () => import('@asyncapistudio/plugin-google-analytics'), // I didn't test it, so I don't know if this works, but should
      { ssr: false }
    )
    
    clientEvents.on('page:render', () => DynamicPlugin.init(window));
    ...
    ```
    
    So we have hardcoded name of the plugin in source file. How to get external plugins from `config.plugins` array? To be honest I don't know, because then we must to do:
    
    ```js
    const pluginsName = ... // here we must load the names of plugins from `config.plugins` array, but we cannot use filesystem, because data from it will be lost in client side, maybe some logic in `getInitialProps` will help
    
    for (const pluginName of pluginsName) {
      const DynamicPlugin = dynamic(
        () => import(pluginName), // I didn't test it, so I don't know if this works, but should
        { ssr: false }
      );
      clientEvents.on('page:render', () => ... bind page:render event from plugin);
    }
    ```
    
- Execute `page:render` event.

Let me know if you understand the problem. I really don't know how to solve it the easiest way, because we're trying to bind the filesystem to the client side which is hard to do, I can say it is impossible. I don't see even very complicated solution at the moment for this problem 😢 

I will try to spend some time on this problem, but I don't give guarantee that I will be able to solve something.

Reply:
@AceTheCreator Hi! No worries, we know that you also study :) It's an open source so we also know that people works in the free time :) Please also don't waste your time on this issue, because it's really hard to make it in the good way. You should focus on your GSOC project :)